pyOCD/rtos/rtx5.py

```diff
@@ -135,57 +132,78 @@ def readCoreRegistersRaw(self, reg_list):
         reg_list = [register_name_to_index(reg) for reg in reg_list]
         reg_vals = []
 
-        inException = self._get_ipsr() > 0
         isCurrent = self._thread.is_current
+        inException = isCurrent and self._get_ipsr() > 0
 
         # If this is the current thread and we're not in an exception, just read the live registers.
         if isCurrent and not inException:
             return self._parent.readCoreRegistersRaw(reg_list)
 
-        sp = self._thread.get_stack_pointer()
+        # Because of above tests, from now on, inException implies isCurrent;
+        # we are generating the thread view for the RTOS thread where the
+        # exception occurred; the actual Handler Mode thread view is produced
+        # by HandlerModeThread
+        if inException:
+            # Reasonable to assume PSP is still valid
+            sp = self._parent.readCoreRegister('psp')
+        else:
+            sp = self._thread.get_stack_pointer()
 
         # Determine which register offset table to use and the offsets past the saved state.
         realSpOffset = 0x40
         realSpExceptionOffset = 0x20
         table = self.NOFPU_REGISTER_OFFSETS
         if self._has_fpu:
             try:
-                # Read stacked exception return LR.
-                exceptionLR = self._thread.get_stack_frame()
-
-                # Check bit 4 of the saved exception LR to determine if FPU registers were stacked.
-                if (exceptionLR & (1 << 4)) == 0:
+                if inException and not self._parent.core.isVectorCatch():
```

Comment:
Thought: performance issue? This `isVectorCatch()` goes straight to core and bypasses memory cache?